Add status popover design documentation

andybraren opened the pull request:
This PR introduces a new Status Popover pattern that can be applied to resource statuses within tables or object details pages. Clicking a status with blue text reveals a PatternFly 4 popover with additional information and up to two suggested quick actions.

This pattern is particularly useful for statuses like "Provisioning" or "Migrating" that could take multiple minutes to finish, and allows us to provide useful contextual help and recommended actions.

(Note: This extends the existing Resource Statuses documentation and fixes its name in the `README`)

@openshift/team-ux-leads @openshift/team-ux-review

![status-popover-table](https://user-images.githubusercontent.com/9122899/57951623-91fb6c80-78b8-11e9-8b60-733520b3bd75.png)
![status-popover-examples](https://user-images.githubusercontent.com/9122899/58044434-b734e880-7b0d-11e9-9616-1b24ad9ef9d5.png)

matthewcarleton commented on web-console/4.0-designs/status/status.md:
```diff
@@ -47,3 +47,13 @@ Note: Icons should never be animated in a list view.
+
+## Status Popovers
+
+If additional information about a status is available, a popover can be used to provide context and up to two suggested actions. Statuses with popovers should be styled in the same way as inline links to indicate interactivity.
```
```suggestion
If additional information about a status is available, a popover should be used to provide context and up to two suggested actions. Statuses with popovers should be styled in the same way as inline links to indicate interactivity.
```

andybraren replied:
Changed to be more assertive 👍

matthewcarleton commented on web-console/4.0-designs/status/status.md:
```diff
@@ -47,3 +47,13 @@ Note: Icons should never be animated in a list view.
+
+![Status popover in table](img/status-popover-table.png)
+
+Status popovers can include links, progress bars, and dynamically-updated content. Up to two contextual actions can be included in the footer of the popover.
```
should we clarify that there can also be actions included in the body of the popup?

andybraren replied:
I'm not quite sure about this - do you have an example where this would be helpful?

I'd worry somewhat about users confusing hyperlinks with (what should be) inline link buttons for actions if we mix both inside the body. I'm thinking restricting actions to the footer might standardize their interaction and discourage us from putting more than we should into these popovers.

andybraren replied:
I made a small tweak to add an external link icon whenever blue text in the body or footer would bring the user to another site. Blue text within the body would link to other internal pages, and blue "action" text within the footer would behave like... actions.

matthewcarleton replied:
+1 on the external link icon. That definitely helps separate things in the body.
I agree we don't want to over run these with actions.
Thinking about this from an accessibility point of view an action could be more helpful in the context of the content than as a separate action in the footer.

andybraren replied:
I softened the wording to allow for actions within the body "when it makes sense to do so" to accommodate potential use cases that we can't predict right now. We can make our guidance more prescriptive again if this leniency starts making things too inconsistent.

lizsurette commented:
LGTM! @beanh66 - WDTY?

serenamarie125 commented on web-console/4.0-designs/status/status.md:
```diff
@@ -47,3 +47,13 @@ Note: Icons should never be animated in a list view.
+
+## Status Popovers
+
+If additional information about a status is available, a popover should be used to provide context and up to two suggested actions. Statuses with popovers should be styled in the same way as inline links to indicate interactivity.
```
@andybraren why are we limiting to 2?  might it make sense to allow a primary as a link, but then offer a kebab or some other pattern to provide an overflow mechanism?

andybraren replied:
I softened the wording of this to allow "one or more suggested actions" in order to accommodate future use cases.

I haven't personally encountered a situation where more than 2 actions (not links, which could be included in the body) were relevant to a particular status, but I suppose that could happen.

andybraren commented:
@beanh66 I've updated this PR to point to @ncameronbritt's Contextual Help doc and added two links in the introduction and Infopop sections of that doc to this new Status Popover pattern.

I also softened the wording of this Status Popover pattern to be less prescriptive based on the feedback above. We'll see how others use this pattern and tweak as we go.

ncameronbritt commented:
LTGM!
